internal/util/projutil/project_util.go

```diff
@@ -97,3 +88,27 @@ func GetOperatorType() OperatorType {
 	}
 	return OperatorTypeGo
 }
+
+func SaveGopath() string {
+	return os.Getenv(GopathEnv)
+}
+
+func SetGopath(currentGopath string) {
+	var newGopath string
+	cwdInGopath := false
+	wd := MustGetwd()
+	for _, newGopath = range strings.Split(currentGopath, ":") {
+		if strings.HasPrefix(filepath.Dir(wd), newGopath) {
+			cwdInGopath = true
+			break
+		}
+	}
+	if !cwdInGopath {
+		log.Fatalf("check current pkg failed: must run from gopath")
+	}
+	os.Setenv(GopathEnv, newGopath)
+}
+
+func RestoreGopath(oldGopath string) {
```

Comment:
The naming is confusing here. I'd rather call
```Go
os.SetEnv(projutil.GopathEnv, gopath)
````
in `main.go`.

Reply:
I don't think we need to restore GOPATH.
`SetGopath()` or more specifically `os.Setenv` will only change the env for it's own process(and subprocesses which inherit envs). The change won't affect the parent process so we don't need to restore before exiting.

So we can remove this.

Reply:
updated